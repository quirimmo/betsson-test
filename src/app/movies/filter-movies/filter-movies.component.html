<form>
	<div class="row search-movie-container">
		<mat-form-field class="col-md-6 col-12 w-100">
			<input
				id="search-movie-name"
				[formControl]="nameText"
				matInput
				placeholder="Type the name of the movie..."
			/>
		</mat-form-field>
		<mat-form-field class="col-md-6 col-12 w-100">
			<mat-select
				id="seach-movie-genres"
				placeholder="Select the genres..."
				[formControl]="genresSelect"
				multiple
			>
				<mat-option *ngFor="let genre of genres" [value]="genre">{{
					genre
				}}</mat-option>
			</mat-select>
		</mat-form-field>
	</div>
</form>
