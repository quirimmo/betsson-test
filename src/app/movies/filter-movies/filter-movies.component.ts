import { Component, Input, OnInit, OnDestroy } from '@angular/core';
import { FormControl } from '@angular/forms';
import { Subscription } from 'rxjs';
import { ActivatedRoute, Router } from '@angular/router';
import { MoviesRouterService } from '../movies-router.service';

@Component({
	selector: 'filter-movies',
	inputs: ['genres'],
	styleUrls: ['./filter-movies.component.scss'],
	templateUrl: './filter-movies.component.html'
})
export class FilterMoviesComponent implements OnInit, OnDestroy {
	@Input()
	genres: string[];
	genresSelect = new FormControl();
	nameText = new FormControl();
	nameTextChangesSubscriber: Subscription;
	genresChangesSubscriber: Subscription;

	constructor(
		private moviesRouterService: MoviesRouterService,
		private route: ActivatedRoute,
		private router: Router
	) {}

	ngOnInit(): void {
		const searchName: string =
			this.moviesRouterService.getSearchNameParam() || '';
		const searchGenres: string[] =
			this.moviesRouterService.getSearchGenresParam() || this.genres;

		this.nameTextChangesSubscriber = this.nameText.valueChanges.subscribe(
			this.onChangeSearchName.bind(this)
		);
		this.genresChangesSubscriber = this.genresSelect.valueChanges.subscribe(
			this.onChangeSearchGenres.bind(this)
		);

		this.nameText.setValue(searchName);
		this.genresSelect.setValue(searchGenres);
	}

	ngOnDestroy(): void {
		this.nameTextChangesSubscriber.unsubscribe();
		this.genresChangesSubscriber.unsubscribe();
	}

	public onChangeSearchName(): void {
		this.moviesRouterService.appendSearchNameParam(this.nameText.value);
	}

	public onChangeSearchGenres(): void {
		this.moviesRouterService.appendSearchGenresParam(this.genresSelect.value);
	}
}
