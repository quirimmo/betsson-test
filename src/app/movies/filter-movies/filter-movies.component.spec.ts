import { TestBed, async } from '@angular/core/testing';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { FilterMoviesComponent } from './filter-movies.component';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import {
	MatFormFieldModule,
	MatInputModule,
	MatSelectModule
} from '@angular/material';

const providers = [];
const imports = [
	BrowserAnimationsModule,
	FormsModule,
	ReactiveFormsModule,
	MatSelectModule,
	MatInputModule,
	MatFormFieldModule
];
const declarations = [FilterMoviesComponent];

describe('FilterMovies Component', () => {
	let componentFixture, componentInstance;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			declarations,
			providers,
			imports
		}).compileComponents();
		componentFixture = TestBed.createComponent(FilterMoviesComponent);
		componentInstance = componentFixture.debugElement.componentInstance;
		componentInstance.genres = ['genre1'];
	}));

	it('should create the component', () => {
		expect(componentFixture).toBeTruthy();
	});

	it('should match the snapshot', () => {
		expect(componentFixture).toMatchSnapshot();
	});

	describe('OnInit', () => {
		beforeEach(async () => {
			componentFixture.detectChanges();
			await componentFixture.whenStable();
		});
		it('should init the name text', () => {
			expect(componentInstance.nameText.value).toEqual('');
		});

		it('should init the genres select', () => {
			expect(componentInstance.genresSelect.value).toEqual(['genre1']);
		});
	});
});
