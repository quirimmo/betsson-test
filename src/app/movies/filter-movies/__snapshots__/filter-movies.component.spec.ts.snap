// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterMovies Component should match the snapshot 1`] = `
<filter-movies
  genres={[Function Array]}
  genresSelect={[Function FormControl]}
  nameText={[Function FormControl]}
>
  <form
    novalidate=""
  >
    <div
      class="row search-movie-container"
    >
      <mat-form-field
        class="col-md-6 col-12 w-100 mat-form-field ng-tns-c3-0 mat-primary"
      >
        <div
          class="mat-form-field-wrapper"
        >
          <div
            class="mat-form-field-flex"
          >
            
            
            <div
              class="mat-form-field-infix"
            >
              <input
                class="mat-input-element mat-form-field-autofill-control"
                matinput=""
                placeholder="Type the name of the movie..."
              />
              <span
                class="mat-form-field-label-wrapper"
              >
                
              </span>
            </div>
            
          </div>
          
          <div
            class="mat-form-field-subscript-wrapper"
          >
            
            
          </div>
        </div>
      </mat-form-field>
      <mat-form-field
        class="col-md-6 col-12 w-100 mat-form-field ng-tns-c3-1 mat-primary"
      >
        <div
          class="mat-form-field-wrapper"
        >
          <div
            class="mat-form-field-flex"
          >
            
            
            <div
              class="mat-form-field-infix"
            >
              <mat-select
                class="mat-select ng-tns-c4-2"
                multiple=""
                placeholder="Select the genres..."
                role="listbox"
              >
                <div
                  aria-hidden="true"
                  cdk-overlay-origin=""
                  class="mat-select-trigger"
                >
                  <div
                    class="mat-select-value"
                  >
                    
                    
                  </div>
                  <div
                    class="mat-select-arrow-wrapper"
                  >
                    <div
                      class="mat-select-arrow"
                    />
                  </div>
                </div>
                
              </mat-select>
              <span
                class="mat-form-field-label-wrapper"
              >
                
              </span>
            </div>
            
          </div>
          
          <div
            class="mat-form-field-subscript-wrapper"
          >
            
            
          </div>
        </div>
      </mat-form-field>
    </div>
  </form>
</filter-movies>
`;
